tests/FromCommunity/IssueWithQueryArrayTest.php

```diff
@@ -12,62 +12,183 @@ final class IssueWithQueryArrayTest extends TestCase
+        $validator = (new ValidatorBuilder())->fromYaml($this->makeArrayYaml('form', 'boolean', 'int32'))->getServerRequestValidator();
         $validator->validate($this->makeRequest('form', 'boolean'));
         $this->addToAssertionCount(1);
     }
 
     public function testConvertFormIntegerArrayError() : void
     {
         $this->expectExceptionMessage('Value "id1,id2,id3" for argument "id" is invalid for Request [get /users]');
-        $validator = (new ValidatorBuilder())->fromYaml($this->makeYaml('form', 'integer', 'int32'))->getServerRequestValidator();
+        $validator = (new ValidatorBuilder())->fromYaml($this->makeArrayYaml('form', 'integer', 'int32'))->getServerRequestValidator();
         $validator->validate($this->makeRequest('form', 'string'));
         $this->addToAssertionCount(1);
     }
 
     public function testConvertSpaceIntegerArray() : void
     {
-        $validator = (new ValidatorBuilder())->fromYaml($this->makeYaml('spaceDelimited', 'integer', 'int32'))->getServerRequestValidator();
+        $validator = (new ValidatorBuilder())->fromYaml($this->makeArrayYaml('spaceDelimited', 'integer', 'int32'))->getServerRequestValidator();
         $validator->validate($this->makeRequest('spaceDelimited', 'integer'));
         $this->addToAssertionCount(1);
     }
 
     public function testConvertPipeIntegerArray() : void
     {
-        $validator = (new ValidatorBuilder())->fromYaml($this->makeYaml('pipeDelimited', 'integer', 'int32'))->getServerRequestValidator();
+        $validator = (new ValidatorBuilder())->fromYaml($this->makeArrayYaml('pipeDelimited', 'integer', 'int32'))->getServerRequestValidator();
         $validator->validate($this->makeRequest('pipeDelimited', 'integer'));
         $this->addToAssertionCount(1);
     }
 
-    protected function makeYaml(string $style, string $type, string $format) : string
+    public function testConvertSingleLayerDeepObject() : void
+    {
+        $yaml      = /** @lang yaml */
+            <<<YAML
+openapi: 3.0.0
+info:
+  title: Product import API
+  version: '1.0'
+servers:
+  - url: 'http://localhost:8000/api/v1'
+paths:
+  /users:
+    get:
+      parameters:
+        - in: query
+          name: id
+          required: true
+          style: deepObject
+          explode: true
+          schema:
+            type: object
+            properties:
+              before:
+                type: integer
+                format: int32
+              after:
+                type: integer
+                format: int32
+      responses:
+        '200':
+          description: A list of users
+YAML;
+        $validator = (new ValidatorBuilder())->fromYaml($yaml)->getServerRequestValidator();
+        $validator->validate($this->makeRequest('deepObject', 'integer'));
+        $this->addToAssertionCount(1);
+    }
+
+    public function testConvertMultiLayerDeepObject() : void
```

Comment:
Can you please add a error test where only the deepest element is not valid

Reply:
Updated test to break on deepest element